portfolio/src/main/webapp/script.js

```diff
@@ -12,42 +12,108 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/** 
- * FUNCTIONS 
+/**
+ * Ensures that the stated elements are present. Note that elements maybe an
+ * HTMLCollection, NodeList or an HTML element. Hence, the stated elements are considered
+ * to be present if all HTMLCollection and NodeList have length > 0 and all HTML elements
+ * are non-null.
+ * @param {...(NodeList | HTMLCollection | HTML element)} elements Elements to be checked.
+ * @throws Will throw if one of the elements is deemed to be not present
  */
+function ensureNonNull(... elements) {
+  for (element of elements) {
+    isEmpty = (element instanceof HTMLCollection || element instanceof NodeList) &&
+      element.length === 0;
+    isNull = element === null;
+
+    if (isEmpty || isNull) {
+      throw new Error("Missing desired element");
+    }
+  }
+}
 
-/* NAV */
 const mainViews = document.getElementsByClassName('main-view');
 const dropdownMenu = document.getElementById('dropdown-menu');
+try {
+  ensureNonNull(mainViews, dropdownMenu);
+} catch (err) {
+  console.error(err);
+}
 
-// hide dropdown menu and show main view elements
+// Hides dropdown menu and shows main view elements.
 const hideMenuShowMain = () => {
   dropdownMenu.style.display = 'none';
   for (let mainView of mainViews) {
     mainView.style.display = 'flex';
   }
 }
 
- /**
-  * ADD EVENT LISTENERS
-  */
-
-/* NAV */
-const menuIcon = document.getElementById('menu-icon');
-// hide main view elements and display dropdown menu
-menuIcon.onclick = () => {
+// Hides main view elements and shows dropdown menu.
+const hideMainShowMenu = () => {
   for (let mainView of mainViews) {
-   mainView.style.display = 'none';
+    mainView.style.display = 'none';
   }
   dropdownMenu.style.display = 'block';
 }
 
-const closeButton = document.getElementById('close-button');
-// hide dropdown menu and show main view
-closeButton.onclick = hideMenuShowMain;
+/**
+ * Adds listener to dropdown menu, elements that belong to main view 
+ * and elements in the dropdown menu to enable the user to toggle between them. 
+ * To view the dropdown menu, the user has to click on the menu icon.
+ * To view the main view after opening the dropdown, the user has to click close
+ * or select an element in the dropdown menu.
+ */
+const setupToggleMainAndMenu = () => {
+  const closeButton = document.getElementById('close-button');
+  try {
```

Comment:
Glad to see the defensive code.

One comment that I have is that (won't block your approval): There seems to be quite some boilerplate with the current implementation, a even simpler way to do it would be that the ensureNonNull(or call it isPresent) could simply return a true/false instead of throwing an error and then we try/catch, because we don't actually have handling on the case the element is null.

But this is just an optional comment, and I'll still let this PR pass to not block you any further.

Reply:
I see! Thanks for the advice. (: If I choose to return a true/false value, should I console.log the result?

Reply:
That won't be needed :)

Reply:
I see! Pardon my ignorance, but in this case, how would I know when an error occurs?

Reply:
I wouldn't say this is an serious error, the general guideline is that we should only try-catch if we have a meaningful way of handling them, which I don't think we need right now. However, if you do want to know when such error happens, console.log will do, but in production there should be some other ways for it, as you may not want everyone to see the error messasge.

Reply:
Yup! I agree that console.log isn't the best once the version goes public. Thanks for the clarification.